pydantic/_internal/_generate_schema.py

```diff
@@ -2507,8 +2478,112 @@ class _Definitions:
     """Keeps track of references and definitions."""
 
     def __init__(self) -> None:
-        self.seen: set[str] = set()
-        self.definitions: dict[str, core_schema.CoreSchema] = {}
+        self._recursively_seen: set[str] = set()
```

Comment:
`_Definitions` should be probably refactored to its own file. This _generate_schema is a very bloated one